app/controllers/api/v1/users/by_github_controller.rb

```diff
@@ -0,0 +1,11 @@
+class Api::V1::Users::ByGithubController < Api::V1::ApiController
+  before_action :doorkeeper_authorize!
+
+  def show
+    if params[:q] && user = User.find_by(git_hub: params[:q])
+      render json: user, serializer: UserSerializer, status: 200
+    else
+      render json: "User not found with github username #{params[:q]}", status: 404
+    end
```

Comment:
JSON should always be an object or an array. Just rendering a string isn't really JSON. In fact, it's not clear to me what is actually in the body of this response.

Modify this to be something like 
```
render json: { error: "User not found...." }
```